src/components/cc-domain-management/cc-domain-management.stories.js

```diff
@@ -0,0 +1,417 @@
+ * @typedef {import('./cc-domain-management.types.js').DomainStateMarkingPrimary} DomainStateMarkingPrimary
+ * @typedef {import('./cc-domain-management.types.js').DomainManagementDnsInfoStateLoaded} DomainManagementDnsInfoStateLoaded
+ * @typedef {import('./cc-domain-management.types.js').DomainManagementDnsInfoStateLoading} DomainManagementDnsInfoStateLoading
+ * @typedef {import('./cc-domain-management.types.js').DomainManagementDnsInfoStateError} DomainManagementDnsInfoStateError
+ * @typedef {import('./cc-domain-management.types.js').DomainManagementListStateLoaded} DomainManagementListStateLoaded
+ * @typedef {import('./cc-domain-management.types.js').DomainManagementListStateLoading} DomainManagementListStateLoading
+ * @typedef {import('./cc-domain-management.types.js').DomainManagementListStateError} DomainManagementListStateError
+ * @typedef {import('./cc-domain-management.types.js').DomainManagementFormStateIdle} DomainManagementFormStateIdle
+ * @typedef {import('./cc-domain-management.types.js').DomainManagementFormStateAdding} DomainManagementFormStateAdding
+ * @typedef {import('./cc-domain-management.js').CcDomainManagement} CcDomainManagement
+ */
+
+const conf = {
+  component: 'cc-domain-management',
+};
+
+const randomDomainName = randomString(7).toLowerCase();
+
+/** @type {DomainStateIdle[]} */
+const baseDomains = [
+  { type: 'idle', fqdn: 'www.example.com', isPrimary: true },
+  { type: 'idle', fqdn: 'example.com', isPrimary: false },
+  { type: 'idle', fqdn: 'blog.example.com', isPrimary: false },
+  { type: 'idle', fqdn: '*.example.com', isPrimary: false },
+  { type: 'idle', fqdn: 'example.com/api', isPrimary: false },
+  { type: 'idle', fqdn: 'example.org', isPrimary: false },
+  { type: 'idle', fqdn: 'www.example.org/blog', isPrimary: false },
+  { type: 'idle', fqdn: 'blog.example.org', isPrimary: false },
+  { type: 'idle', fqdn: 'perso.example.org', isPrimary: false },
+  { type: 'idle', fqdn: `${randomDomainName}.cleverapps.io`, isPrimary: false },
+];
+
+/** @type {DomainStateIdle[]} */
+const longBaseDomains = [
+  { type: 'idle', fqdn: 'very-very-very-very-very-very-very-very-very-long.example.com', isPrimary: false },
+  { type: 'idle', fqdn: 'very-very-very-very-very-very-very-very-very-long.example.com', isPrimary: false },
+  { type: 'idle', fqdn: 'very-very-very-very-very-very-very-very-very-long.cleverapps.io', isPrimary: false },
+];
+
+/** @type {DomainStateIdle} */
+const httpOnlyDomain = { type: 'idle', fqdn: `subdomain.${randomDomainName}.cleverapps.io`, isPrimary: false };
+
+/** @type {DomainManagementDnsInfoStateLoaded} */
+const baseDnsInfo = {
+  type: 'loaded',
+  cnameValue: 'example.com.',
+  aValues: [
+    '93.184.216.34',
+    '93.184.216.34',
+    '93.184.216.34',
+    '93.184.216.34',
+    '93.184.216.34',
+    '93.184.216.34',
+  ],
+};
+
+export const defaultStory = makeStory(conf, {
+  items: [{
```

Comment:
**question** : I am not sure to understand why/how you do not need to put `domainFormState` in the items for some stories (here for the *defaultStory*, but also in *dataLoadedWithMoreDomains*, *empty*, etc.) and there is still a form in those stories ?

Reply:
The `domainFormState` property has a default value (empty values for every form control + no error).
This means we only have to set the `domainFormState` when we want the component to be in a different state than empty form controls.

Your comment was on point, I was setting `domainFormState` to the same value as the default in many stories where it wasn't needed.
It should be fixed now (only stories with form values / form errors should have items with `domainFormState` properties).

Reply:
I go it, thanks for the explanations ! :pray: